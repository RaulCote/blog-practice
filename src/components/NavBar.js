import React from 'react';
import PropTypes from 'prop-types';
import { RippleButton } from '../ui/buttons';
import { ToggleThemeButton } from '../ui/themeToggleIcon';

const NavBar = ({ toggleMode, isDarkMode }) => {

  const toggleToolTip = isDarkMode 
    ? 'Change to Light Theme' 
    : 'Change to Dark Theme';

  return (
  <>
    <ToggleThemeButton 
      onClick={toggleMode} 
      title={toggleToolTip} 
    />
    <RippleButton 
      onClick={toggleMode} 
      width={'100px'} 
      height={'30px'}
    >
        Ripple
    </RippleButton>
  </>
  )
};

NavBar.propTypes = {
  toggleMode: PropTypes.func.isRequired
};

export default NavBar;
