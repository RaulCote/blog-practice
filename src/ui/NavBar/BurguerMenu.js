import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { MEDIA_QUERY_MOBILE_LIMIT } from '../../ui/variables';

const BurguerMenuStyled = styled.div`
  display: none;

  ${MEDIA_QUERY_MOBILE_LIMIT} {
    display: flex;
    justify-content: center;
    flex-wrap: wrap;
    position: absolute;
    top: 0px;
    left: 0px;
    z-index: 2;

    input {
      display: none;
    }

    label {
      cursor: pointer;
      width: 100%;
      height: 100%;
      padding: 21px 21px;
    }

    #burguer-bar {
      width: 22px;
      margin: 6px 0px;
      height: 2px;
      background-color: ${props => props.theme.colors.titles};
      transition: transform 1s cubic-bezier(0.23, 1, 0.32, 1),
        color 1s cubic-bezier(0.23, 1, 0.32, 1);
      top: 50%;
      left: 50%;
      -webkit-transition: -webkit-transform 1s cubic-bezier(0.23, 1, 0.32, 1),
        color 1s cubic-bezier(0.23, 1, 0.32, 1);
      -webkit-transform: translateY(-3.75px) translateZ(0);
      transform: translateY(-3.75px) translateZ(0);
      &:last-child {
        -webkit-transform: translateY(3.75px) translateZ(0);
        transform: translateY(3.75px) translateZ(0);
      }
    }

    input:checked + label {
      #burguer-bar {
        transform: rotate(45deg) translateZ(0);
        top: 60%;
        position: absolute;
        &:last-child {
          transform: rotate(-45deg) translateZ(0);
          top: 60%;
          position: absolute;
        }
      }
    }
  }
`;

const BurguerMenu = ({ isResponsiveMenuOpen, onToggle }) => (
  <BurguerMenuStyled>
    <input
      type={'checkbox'}
      id={'burguer-check'}
      checked={isResponsiveMenuOpen}
      onChange={onToggle}
    />
    <label htmlFor={'burguer-check'}>
      <div id={'burguer-bar'} />
      <div id={'burguer-bar'} />
    </label>
  </BurguerMenuStyled>
);

BurguerMenu.propTypes = {
  isResponsiveMenuOpen: PropTypes.bool.isRequired,
  onToggle: PropTypes.func.isRequired,
};

export default BurguerMenu;
