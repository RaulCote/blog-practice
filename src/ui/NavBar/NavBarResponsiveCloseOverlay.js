import React from 'react';
import styled from 'styled-components';

const OverlayStyled = styled.div`
  width: 20vh;
  height: 100vh;
  position: fixed;
  top: 0;
  right: 0;
  background-color: transparent;
`;

const NavBarResponsiveCloseOverlay = ({ isResponsiveMenuOpen, onClick }) => {
  if (!isResponsiveMenuOpen) return null;

  return <OverlayStyled onClick={onClick} />;
};

export default NavBarResponsiveCloseOverlay;
