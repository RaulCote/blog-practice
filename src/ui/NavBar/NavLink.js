import React, { useContext } from 'react';
import PropTypes from 'prop-types';
import styled, { ThemeContext } from 'styled-components';
import { Link } from 'gatsby';

export const GatsbyLinkStyled = styled(Link)`
  background-color: ${props => props.theme.colors.background};
  color: ${props => props.theme.colors.textColor};
  cursor: pointer;
  height: ${props => props.height};
  outline: none;
  overflow: hidden;
  position: relative;
  text-decoration: none;
  width: ${props => props.width};
  display: flex;
  justify-content: center;
  align-items: center;
  border-top: 4px solid transparent;
  transition-duration: 0.2s;

  &:hover {
    background-color: #e4e6e9;
  }

  &:active {
    background-color: #d7eaef;
  }
`;

GatsbyLinkStyled.propTypes = {
  activeStyle: PropTypes.shape({
    borderTop: PropTypes.string,
    fontWeight: PropTypes.string,
    textShadow: PropTypes.string,
  }),
  height: PropTypes.string.isRequired,
  width: PropTypes.string.isRequired,
};

const NavLink = ({
  to,
  width,
  height,
  children,
  testId,
  isResponsiveMenuOpen,
}) => {
  const theme = useContext(ThemeContext);

  return (
    <GatsbyLinkStyled
      activeStyle={
        !isResponsiveMenuOpen
          ? {
              borderTop: `4px solid ${theme &&
                theme.colors &&
                theme.colors.textColor}`,
              fontWeight: 'bold',
              textShadow: '0 0 1px black',
            }
          : {}
      }
      data-testid={testId}
      height={height}
      to={to}
      width={width}
    >
      {children}
    </GatsbyLinkStyled>
  );
};

NavLink.propTypes = {
  children: PropTypes.node.isRequired,
  height: PropTypes.string,
  isResponsiveMenuOpen: PropTypes.bool.isRequired,
  to: PropTypes.string.isRequired,
  width: PropTypes.string,
};

NavLink.defaultProps = {
  width: '100%',
  height: '100%',
};

export default NavLink;
