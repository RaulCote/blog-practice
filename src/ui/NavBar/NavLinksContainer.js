import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';
import { GatsbyLinkStyled } from './NavLink';
import {
  MARGIN_CONTAINER,
  MAX_WIDTH_CONTAINER,
  MEDIA_QUERY_MOBILE_FROM,
  MEDIA_QUERY_MOBILE_LIMIT,
} from '../../ui/variables';

const NavLinksContainer = styled.div`
  background-color: ${props => props.theme.colors.background};
  // box-shadow: 0 0 3px 1px ${props => props.theme.colors.shadowColor};

  ${MEDIA_QUERY_MOBILE_FROM} {
    max-width: ${MAX_WIDTH_CONTAINER};
    width: 70%;
    margin: ${MARGIN_CONTAINER};
  }

  ${MEDIA_QUERY_MOBILE_LIMIT} {
    width: 0%;
    height: 100vh;
    transition-duration: 0.2s;
    display: flex;
    flex-direction: column;

    ${GatsbyLinkStyled} {
      color: transparent;
      display: none;
    }

    ${props =>
      props.isResponsiveMenuOpen &&
      css`
        border-top: 4px solid ${props => props.theme.colors.textColor};
        position: relative;
        height: 100vh;
        box-shadow: 0 0 3px 1px ${props => props.theme.colors.shadowColor};
        width: 80%;
        z-index: 1;

        ${GatsbyLinkStyled} {
          display: flex;
          color: ${props => props.theme.colors.textColor};
          font-weight: bold;
          padding-left: 5rem;
          justify-content: flex-start;
        }
      `}
  }


  height: 8vh;
  align-items: center;
  display: flex;
  justify-content: space-evenly;
  color: ${props => props.theme.colors.textColor};
`;

NavLinksContainer.propTypes = {
  isResponsiveMenuOpen: PropTypes.bool.isRequired,
};

export default NavLinksContainer;
