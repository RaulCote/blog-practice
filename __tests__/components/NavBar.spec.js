import React from 'react';
import renderer from 'react-test-renderer';
import NavBar from '../../src/components/NavBar';
import { render } from '@testing-library/react';
import { renderLightTheme } from '../../utils/tests/themeContext';

const NavBarComponent = () => {
  const toggleMode = jest.fn();

  return <NavBar siteTitle="Default Starter" toggleMode={toggleMode} />;
};

describe('NavBar', () => {
  it('renders correctly', () => {
    const tree = renderLightTheme(<NavBarComponent />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
