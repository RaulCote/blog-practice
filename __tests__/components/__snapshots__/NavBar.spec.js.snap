// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar renders correctly 1`] = `
.c4 {
  background-color: #F3F5F7;
  color: #2E3440;
  cursor: pointer;
  height: 100%;
  outline: none;
  overflow: hidden;
  position: relative;
  -webkit-text-decoration: none;
  text-decoration: none;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-top: 4px solid transparent;
  -webkit-transition-duration: 0.2s;
  transition-duration: 0.2s;
  font-weight: bold;
  opacity: 0.9;
}

.c4:hover {
  opacity: 1;
  color: #1B1D22;
  background-color: #E9EDF0;
}

.c1 {
  display: none;
}

.c2 {
  background-color: #F3F5F7;
  height: 8vh;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: space-evenly;
  -webkit-justify-content: space-evenly;
  -ms-flex-pack: space-evenly;
  justify-content: space-evenly;
  color: #2E3440;
}

.c5 {
  background-color: #5E81AC;
  border-radius: 4%;
  border: 0px solid transparent;
  color: #2E3440;
  cursor: pointer;
  font-size: 0.85em;
  font-weight: 600;
  outline: none;
  overflow: hidden;
  padding: 0.5em;
  position: relative;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 1em;
  overflow: hidden;
  border-radius: 0%;
  background-color: #F3F5F7;
  position: absolute;
  top: 0;
  right: 0;
  height: 8vh;
  width: 10%;
  -webkit-transition-duration: 0.2s;
  transition-duration: 0.2s;
}

.c5:hover {
  opacity: 1;
  color: #1B1D22;
  background-color: #E9EDF0;
}

.c0 {
  height: 8vh;
}

@media screen and (max-width:600px) {
  .c1 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
    -webkit-flex-wrap: wrap;
    -ms-flex-wrap: wrap;
    flex-wrap: wrap;
    position: absolute;
    top: 0px;
    left: 0px;
    z-index: 2;
  }

  .c1 input {
    display: none;
  }

  .c1 label {
    cursor: pointer;
    width: 100%;
    height: 100%;
    padding: 21px 21px;
  }

  .c1 #burguer-bar {
    width: 22px;
    margin: 6px 0px;
    height: 2px;
    background-color: #2E3440;
    -webkit-transition: -webkit-transform 1s cubic-bezier(0.23,1,0.32,1), color 1s cubic-bezier(0.23,1,0.32,1);
    -webkit-transition: transform 1s cubic-bezier(0.23,1,0.32,1), color 1s cubic-bezier(0.23,1,0.32,1);
    transition: transform 1s cubic-bezier(0.23,1,0.32,1), color 1s cubic-bezier(0.23,1,0.32,1);
    top: 50%;
    left: 50%;
    -webkit-transition: -webkit-transform 1s cubic-bezier(0.23,1,0.32,1), color 1s cubic-bezier(0.23,1,0.32,1);
    -webkit-transform: translateY(-3.75px) translateZ(0);
    -webkit-transform: translateY(-3.75px) translateZ(0);
    -ms-transform: translateY(-3.75px) translateZ(0);
    transform: translateY(-3.75px) translateZ(0);
  }

  .c1 #burguer-bar:last-child {
    -webkit-transform: translateY(3.75px) translateZ(0);
    -webkit-transform: translateY(3.75px) translateZ(0);
    -ms-transform: translateY(3.75px) translateZ(0);
    transform: translateY(3.75px) translateZ(0);
  }

  .c1 input:checked + label #burguer-bar {
    -webkit-transform: rotate(45deg) translateZ(0);
    -ms-transform: rotate(45deg) translateZ(0);
    transform: rotate(45deg) translateZ(0);
    top: 60%;
    position: absolute;
  }

  .c1 input:checked + label #burguer-bar:last-child {
    -webkit-transform: rotate(-45deg) translateZ(0);
    -ms-transform: rotate(-45deg) translateZ(0);
    transform: rotate(-45deg) translateZ(0);
    top: 60%;
    position: absolute;
  }
}

@media screen and (min-width:601px) {
  .c2 {
    max-width: 960px;
    margin: auto;
  }
}

@media screen and (max-width:600px) {
  .c2 {
    width: 0%;
    height: 100vh;
    -webkit-transition-duration: 0.2s;
    transition-duration: 0.2s;
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }

  .c2 .c3 {
    color: transparent;
    display: none;
  }
}

@media screen and (max-width:600px) {
  .c5 {
    width: 20%;
  }
}

<nav
  className="c0"
>
  <div
    className="c1"
  >
    <input
      checked={false}
      id="burguer-check"
      onChange={[Function]}
      type="checkbox"
    />
    <label
      htmlFor="burguer-check"
    >
      <div
        id="burguer-bar"
      />
      <div
        id="burguer-bar"
      />
    </label>
  </div>
  <div
    className="c2"
  >
    <a
      className="c3 c4"
      height="100%"
      href="/"
      width="100%"
    >
      Default Starter
    </a>
    <a
      className="c3 c4"
      height="100%"
      href="/about"
      width="100%"
    >
      About
    </a>
    <a
      className="c3 c4"
      height="100%"
      href="/work"
      width="100%"
    >
      Work
    </a>
  </div>
  <button
    className="sc-AykKI c5"
    data-testid="ripple-button"
    onClick={[Function]}
  >
    Dark
  </button>
</nav>
`;
